docs/RFCS/20220112_bundle_js_with_esbuild.md

```diff
@@ -0,0 +1,715 @@
+without having to recompile and re-bundle an application, and rely on
+rollup.js and esbuild (respectively) to produce optimized production
+builds. With this knowledge, it's clear that neither vite nor snowpack
+can ever produce production builds faster than the standalone bundlers
+they depend on. And while their "bundle-less" development models offer
+extremely attractive guarantees, the bootstrapping process is
+surprisingly slow: for each file used in production (including all
+dependencies in `node_modules/`), an ES Module equivalent must be
+produced and cached on-disk. This also makes development builds
+extremely different from production builds, since different compilers
+are used.
+
+Tools like snowpack are useful to track moving forward, but their
+reliance on browser-native ES Modules makes failure patterns less clear.
+For an infrastructure-level project like CockroachDB, that risk isn't
+appropriate yet.
+
+### Impacts of Inaction
+
+There's no stability risks to taking no action here. Webpack builds are
+slow, but aren't actively causing production issues. Continuing with
+webpack does however mean that developer time and CPU time will be used
+inefficiently. Inaction will likely limit the rate at which CockroachDB
+features are developed, iterated on, etc.
+
+# Explain it to folk outside of your team
+
+We currently use a project called [webpack](https://webpack.js.org/) to
+build our web application, but webpack is pretty slow. It takes a few
+minutes for it to build the DB Console in its various formats, and that
+time is typically seen as unusable by devs working in CockroachDB.
+Switching web app build systems introduces some small risks, but can
+save several minutes per build per dev per day.
+
+This change would be transparent to end-users and CockroachDB cluster
+operators, as the end-result of this build is embedded in the
+`cockroach` binaries they use. This includes:
+
+- No new concepts to introduce to end-users
+- No change in thinking for end-users with respect to their applications
+  or mental models
+- No new error messages or deprecation warnings
+- No impact to clusters created before this change (a new binary
+  includes the new build system output, so upgrades are atomic)
+
+# Unresolved questions
+
+There's very few unresolved questions at this point. Initial
+[protoyping](https://github.com/sjbarag/cockroach/commit/f9a17d4dda9ba6c964e87e5fbb817eb6e601c366)
+stopped just short of support for inlined assets (a la
+`raw-loader!../path/to/file.ext` in webpack) and hadn't quite worked out
+font bundling, but those are significantly smaller concerns.  That
+Parcel 2 benchmarking was slower than the existing webpack build is a
+bit surprising and needs a bit more investigation. Similarly, snowpack's
+maintenance patterns need investigation before it can be considered a
+valid option in the future.
+
+Finally, integration with bazel and make need to be confirmed.  Given
```

Comment:
Integrating only with Bazel and forgoing `make` is a sensible choice if it saves any time, but I don't know if it's conceptually possible.